core/frontend/src/render/webgl/BranchUniforms.ts

```diff
@@ -103,19 +103,12 @@ export class BranchUniforms {
 
   public pushBranch(branch: Branch): void {
     desync(this);
-    this._stack.pushBranch(branch);
-    const disableClipStyle = branch.disableClipStyle || this.top.disableClipStyle;
-    const vp = IModelApp.viewManager.selectedView;
-    if (vp) {
 
-      /** If the branch has disableClipStyle set, we set these alphas to 0, therefore disabling the clip style.
-      * Otherwise, we leave it to the viewport's clip style.
-      */
-      const style = vp.view.displayStyle.settings.clipStyle;
-      this.clipStack.insideColor.alpha = disableClipStyle ? 0 : (style.insideColor ? 1 : 0);
-      this.clipStack.outsideColor.alpha = disableClipStyle ? 0 : (style.outsideColor ? 1 : 0);
-      this.clipStack.intersectionStyle.alpha = disableClipStyle ? 0 : (style.intersectionStyle ? style.intersectionStyle.width : 0);
-    }
+    if (this.top.disableClipStyle)
+      branch.disableClipStyle = true;
```

Comment:
that's been removed, and the clip style is now being set with `setClipStyle(this.top.disableClipStyle)` after the branch is pushed, meaning the newly created `BranchState` is being used to determine the clip style and therefore, `disableClipStyle` will be fully inherited without modifying the input `Branch`.

Reply:
Ex:
```
A: disableClipStyle = true
  B
  C
```

Both B and C should draw with clip style disabled.
But in your implementation, C will draw with clip style enabled, because you re-enable it when you pop B.